migrate.js

```diff
@@ -98,30 +97,29 @@ exports.version = '0.4.0'
 
 exports.manifest = {
   start: 'sync',
+  doesOldLogExist: 'sync',
 }
 
-exports.init = function init(sbot, config, newLogMaybe) {
+exports.init = function init(sbot, config) {
   const oldLogExists = makeFileExistsObv(oldLogPath(config.path))
 
   let started = false
   let hasCloseHook = false
   let retryPeriod = 250
-  let retryTimer
-  let liveDebugTimer
+  let drainAborter = null
 
   function oldLogMissingThenRetry(fn) {
     if (!hasCloseHook) {
       sbot.close.hook(function (fn, args) {
-        clearTimeout(retryTimer)
-        clearInterval(liveDebugTimer)
+        if (drainAborter) drainAborter.abort()
         fn.apply(this, args)
       })
       hasCloseHook = true
     }
     oldLogExists.set(fileExists(oldLogPath(config.path)))
     if (oldLogExists.value === false) {
       retryPeriod = Math.min(retryPeriod * 2, 8000)
-      retryTimer = setTimeout(fn, retryPeriod)
+      setTimeout(fn, retryPeriod).unref()
```

Comment:
I should remember about `unref()`, it's in many cases better than keeping a ref to the timer (which forces Node.js to stay up)